rs/messaging/src/routing/stream_handler.rs

```diff
@@ -963,12 +929,14 @@ impl StreamHandlerImpl {
             }
 
             // Best-effort response to canister hosted by other subnet. May occur
-            // legitimately if the canister was migrated after having timed out the
-            // matching callback.
+            // legitimately if the canister was migrated after the matching callback had
+            // timed out.
             Some(_) if msg.is_best_effort() && matches!(msg, RequestOrResponse::Response(_)) => {
-                self.observe_inducted_message_status(msg_type, LABEL_VALUE_DROPPED);
-                // Cycles were lost.
-                Accept(msg_cycles)
+                self.observe_inducted_message_status(
+                    msg_type,
+                    LABEL_VALUE_RECEIVER_LIKELY_MIGRATED,
+                );
+                Reject(RejectReason::CanisterMigrating, msg)
```

Comment:
This, and the change immediately below are the only change in behavior.

Everything else is refactoring.

Reply:
Based on the comment above, I'm surprised that we ended up putting `Accept` here. Was this a bug or was there some reason that we used to accept here before this PR?

Reply:
Indeed, without refunds this was the right thing to do. Thanks!